### DISCUSSION TITLE
Wayne Dalton thermostat WDTC-20 identified as light switch

### DISCUSSION THREAD

**Discussion by alexeip0** [zwave-js/zwave-js#6817] [Request Support / Investigate Issue] [answered]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

What is happening?

Using Home Assistant Green with ZST39 and Zwave JS UI.

Successfully including WDTC-20. But it shows up as a light switch.

What did you expect to happen instead?

To be identified as a Thermostat.

  1. Go to Zwave JS UI
  2. Click on button at the bottom right, choose "Manage nodes" -> Inclusion, follow the prompts
  3. Device is successfully included
  4. In Zwave JS UI listed as "Product" : "Thermostat", but in Settings -> Devices shows up as light bulb
  5. In diagnostics log shows up as         "domain": "light",  "entity_id": "light.thermostat_basic",

Anything else we should know?

First discussed the issue here:
https://community.home-assistant.io/t/thermostat-recognized-as-a-light/725972
was directed to collect driver debug logs and open new issue here.

### Software versions

Driver (node-zwave-js): 12.5.5
Z-Wave JS UI: 9.11.1 (Add-on 3.5.0)
Home Assistant Core: 2024.4.4

### Device information

Manufacturer: Wayne Dalton
Model name: WDTC-20
Node ID: Node 004
Manufacturer ID: 0x0008
Product Type: 0x0007
Product ID: 0x0020

https://devices.zwave-js.io/?jumpTo=0x0008:0x0007:0x0020:0.8
https://opensmarthouse.org/zwavedatabase/1071

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [X] The log includes a re-interview of the problematic device _(if applicable)_.

- [ ] The log includes the problematic interaction with the device _(if applicable)_.

- [X] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

[zwavejs_2024-05-07.log](https://github.com/zwave-js/node-zwave-js/files/15239191/zwavejs_2024-05-07.log)
[zwave_js-7075474b92eb6a6820b231c62e6bf29a-Thermostat-548c65671f05f5f862c7e91ec3adeeda.json](https://github.com/zwave-js/node-zwave-js/files/15239207/zwave_js-7075474b92eb6a6820b231c62e6bf29a-Thermostat-548c65671f05f5f862c7e91ec3adeeda.json)

**Comment by alexeip0** [comment]
Here are some excerpts from the log that might be relevant.

Device seems to report all these commands as supported
```2024-05-07T16:57:45.094Z CNTRLR « [Node 004] node info received
                                  supported CCs:
                                  · Manufacturer Specific
                                  · Thermostat Mode
                                  · Thermostat Setpoint
                                  · Version
                                  · Multilevel Sensor
                                  · Thermostat Operating State
                                  · Indicator
                                  · Binary Switch
```
But then reports version 0 when queried about each command and that command is declared unsupported
```
2024-05-07T16:57:45.435Z DRIVER » [Node 004] [REQ] [SendDataBridge]
                                  │ source node id:   1
                                  │ transmit options: 0x25
                                  │ callback id:      7
                                  └─[VersionCCCommandClassGet]
                                      CC: Thermostat Mode
...
2024-05-07T16:57:45.483Z DRIVER « [Node 004] [REQ] [BridgeApplicationCommand]
                                  │ RSSI: -80 dBm
                                  └─[VersionCCCommandClassReport]
                                      CC:      Thermostat Mode
                                      version: 0
2024-05-07T16:57:45.484Z CNTRLR   [Node 004]   does NOT support CC Thermostat Mode (0x40)
```

Some more info. This device has been working fine with my 14 year old Vera1 that has died recently.

I also tried replacing Vera with Hubitat -- it correctly identified this device as Thermostat and let me choose "Generic Thermostat" profile that worked correctly for the most of the above capabilities (I'm trying HA now because Hubitat was unstable, hanging every few days).

**Comment by AlCalzone** [comment, marked as answer]
Looks like you found the root cause already. Should be an easy fix, will do it tomorrow.
--> #6821

**Comment by alexeip0** [comment]
The fix works. Thank you!
